WordPress/Classes/Networking/TaxonomyServiceRemote.h

```diff
@@ -2,26 +2,52 @@
 
 @class RemotePostCategory;
 @class RemotePostTag;
+@class RemoteTaxonomyPaging;
 
-/*
- Interface for requesting taxonomy such as tags and categories on a site.
+/* Interface for requesting taxonomy such as tags and categories on a site.
  */
 @protocol TaxonomyServiceRemote <NSObject>
 
+/* Create a new category with the site.
+ */
+- (void)createCategory:(RemotePostCategory *)category
+               success:(void (^)(RemotePostCategory *category))success
+               failure:(void (^)(NSError *error))failure;
+
 /* Fetch a list of categories associated with the site.
+ * Note: Requests no paging parameters via the API defaulting the response.
  */
 - (void)getCategoriesWithSuccess:(void (^)(NSArray <RemotePostCategory *> *categories))success
                          failure:(void (^)(NSError *error))failure;
 
-/* Create a new category with the site.
+/* Fetch a list of categories associated with the site with paging.
  */
-- (void)createCategory:(RemotePostCategory *)category
-               success:(void (^)(RemotePostCategory *category))success
-               failure:(void (^)(NSError *error))failure;
+- (void)getCategoriesWithSuccess:(void (^)(NSArray <RemotePostCategory *> *categories))success
+                          paging:(RemoteTaxonomyPaging *)paging
+                         failure:(void (^)(NSError *error))failure;
```

Comment:
I don't think we've ever said anything "officially" but in practice the convention has been to group success and failure blocks at the end of a list of agruments.  `paging` looks a bit odd in the middle but maybe its just me?